var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -60,93 +74,82 @@ class Gcc(Package):
+
+    def configure_args(self):
+        spec = self.spec
+
+        enabled_languages = set(('c', 'c++', 'fortran', 'java', 'objc'))
+
+        if spec.satisfies('@4.7.1:') and sys.platform != 'darwin' and \
+           not (spec.satisfies('@:4.9.3') and 'ppc64le' in spec.architecture):
+            enabled_languages.add('go')
 
         # Generic options to compile GCC
-        options = ["--prefix=%s" % prefix, "--libdir=%s/lib64" % prefix,
-                   "--disable-multilib",
-                   "--enable-languages=" + ','.join(enabled_languages),
-                   "--with-mpc=%s" % spec['mpc'].prefix, "--with-mpfr=%s" %
-                   spec['mpfr'].prefix, "--with-gmp=%s" % spec['gmp'].prefix,
-                   "--enable-lto", "--with-quad"]
-        # Binutils
-        if spec.satisfies('+binutils'):
-            static_bootstrap_flags = "-static-libstdc++ -static-libgcc"
-            binutils_options = [
-                "--with-sysroot=/", "--with-stage1-ldflags=%s %s" %
-                (self.rpath_args, static_bootstrap_flags),
-                "--with-boot-ldflags=%s %s" %
-                (self.rpath_args, static_bootstrap_flags), "--with-gnu-ld",
-                "--with-ld=%s/bin/ld" % spec['binutils'].prefix,
-                "--with-gnu-as",
-                "--with-as=%s/bin/as" % spec['binutils'].prefix
-            ]
-            options.extend(binutils_options)
+        args = [
+            '--libdir={0}/lib64'.format(self.prefix),
+            '--disable-multilib',
+            '--enable-languages=' + ','.join(enabled_languages),
+            '--with-gmp={0}'.format(spec['gmp'].prefix),
+            '--with-mpfr={0}'.format(spec['mpfr'].prefix),
+            '--with-mpc={0}'.format(spec['mpc'].prefix),
+            '--enable-lto',
+            '--with-quad'
+        ]
+
         # Isl
         if 'isl' in spec:
-            isl_options = ["--with-isl=%s" % spec['isl'].prefix]
-            options.extend(isl_options)
+            args.append('--with-isl={0}'.format(spec['isl'].prefix))
 
         if sys.platform == 'darwin':
-            darwin_options = ["--with-build-config=bootstrap-debug"]
-            options.extend(darwin_options)
-
-        build_dir = join_path(self.stage.path, 'spack-build')
-        configure = Executable(join_path(self.stage.source_path, 'configure'))
-        with working_dir(build_dir, create=True):
-            # Rest of install is straightforward.
-            configure(*options)
-            if sys.platform == 'darwin':
-                make("bootstrap")
-            else:
-                make()
-            make("install")
+            args.append('--with-build-config=bootstrap-debug')
+
+        return args
+
+    def build(self, spec, prefix):
+        if sys.platform == 'darwin':
+            make('bootstrap')
+        else:
+            make()
+
+    def install(self, spec, prefix):
+        make('install')
 
         self.write_rpath_specs()
```

Comment:
My fault : those things are well documented (or at least this was my intention) in the code, but they didn't find their way through the "packaging" section of the docs yet